packages/php-wasm/compile/php/Dockerfile

```diff
@@ -1,4 +1,8 @@
-FROM playground-php-wasm:base
+# The Bison version to use to build PHP with.
+# Must be 2.7 for PHP <= 7.3
+# Must be 3 for PHP >= 7.4
+ARG BISON_VERSION
+FROM playground-php-wasm:bison${BISON_VERSION}
```

Comment:
Is there any way to compute the bison version here instead of in build.js?

Reply:
@adamziel If we do that, we can't cache it and we'll be forced to rebuild on every iteration.

I think we should move as much of this calculation into the build JS as possible, then think about re-ordering the dockerfile to make the best possible use of that.

If we can cache the identical steps between different builds, and re-use them, we should be able to get the build to run a lot faster.

Reply:
@adamziel Nope. If Docker considers an `PHP_VERSION` arg that is different from that of a previous build, then the layer-hash will differ and the cache will not be used for that build.

I think that the value we gain from a shorter build is greater than the value we lose from a self-contained dockerfile. Our build system architecture is already relatively highly coupled, and the existing dockerfile already has dependencies it cannot invoke on its own.

I think this still represents a step forward.

Reply:
https://github.com/WordPress/wordpress-playground/blob/1ef26cc9ee6938fb509fd69260a309606729615d/packages/php-wasm/compile/php/Dockerfile#L24-L39

Reply:
>  If Docker considers a PHP_VERSION arg that is different from that of a previous build, then the layer-hash will differ and the cache will not be used for that build.

Using the `--download-only` flag for apt would do the trick here, though. Both Bison 2.7 and Bison 3.0 may then exist in the docker cache as static files, and then a bash `if` could just plug-in the correct version depending on the PHP_VERSION.

> the value we gain from a shorter build 

How much shorter is it, though? Are we talking about minutes or seconds?

> Our build system architecture is already relatively highly coupled, and the existing dockerfile already has dependencies it cannot invoke on its own.

Oh, interesting, I haven't thought of it like that. What would be an example of such a dependency?